workflow-ui/modal-android/src/main/java/com/squareup/workflow1/ui/modal/ModalContainer.kt

```diff
@@ -41,25 +43,46 @@ public abstract class ModalContainer<ModalRenderingT : Any> @JvmOverloads constr
 
   private var dialogs: List<DialogRef<ModalRenderingT>> = emptyList()
 
+  private val parentLifecycleOwner by lazy(NONE) { WorkflowLifecycleOwner.get(this) }
+
   protected fun update(
     newScreen: HasModals<*, ModalRenderingT>,
     viewEnvironment: ViewEnvironment
   ) {
     baseViewStub.update(newScreen.beneathModals, viewEnvironment)
 
+    // // This view should only ever be created by a ViewFactory, which means that well-behaved
+    // // containers should have wired it up with a lifecycle.
+    // val myLifecycle = checkNotNull(WorkflowLifecycleOwner.get(this)) {
+    //   "Expected ModalContainer's parent to set a WorkflowLifecycleOwner on it."
+    // }
```

Comment:
Meant to remove this check completely? If we want to ensure parentLifecyleOwner why make it nullable?

Reply:
Hm, probably.

Reply:
Removed this commented block. This is now part of the contract of `WorkflowLifecycleOwner` and enforced at runtime by it.